parser.go

```diff
@@ -20,22 +20,30 @@ import (
 	"io"
 )
 
+// ErrMissingRequiredNode means the tree is missing nodes required to
+// perform an operation.
+var ErrMissingRequiredNode = errors.New("missing required node")
+
 // TODO(#459): Implement parser
 
+// Tree is the parse tree data structure.
 type Tree[V any] struct {
+	// Root points to the root Node in the parse tree.
 	Root *Node[V]
 }
 
+// Node is the structure for a single node in the parse tree.
 type Node[V any] struct {
 	Parent   *Node[V]
 	Children []*Node[V]
 	Value    V
 	// TODO: Position,Line,Column in original input.
 }
 
-// FIXME: Remove channel
+// FIXME: Remove channel.
```

Comment:
I thought the comment was about changing how Parser to not reaad from the lexemes channel.

When Lexer.Lex() encounters an error it will store an error in Lexer.s.err and close the channel.  The Parser has the lexemes channel instead of the Lexer so Parser can't call Lexer.Err() to find out if there was an error.

Changing NewParser to take a Lexer instead of a channel seemed big enough for a separate PR.

Reply:
Makes sense.